scipy/optimize/_qap.py

```diff
@@ -0,0 +1,776 @@
+    Solve the quadratic assignment problem (approximately).
+
+    This function solves the Quadratic Assignment Problem (QAP) and the
+    Graph Matching Problem (GMP) using the Fast Approximate QAP Algorithm
+    (FAQ) [1]_.
+
+    Quadratic assignment solves problems of the following form:
+
+    .. math::
+
+        \min_P & \ {\ \text{trace}(A^T P B P^T)}\\
+        \mbox{s.t. } & {P \ \epsilon \ \mathcal{P}}\\
+
+    where :math:`\mathcal{P}` is the set of all permutation matrices,
+    and :math:`A` and :math:`B` are square matrices.
+
+    Graph matching tries to *maximize* the same objective function.
+    This algorithm can be thought of as finding the alignment of the
+    nodes of two graphs that minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    edge weight differences.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    A : 2d-array, square
+        The square matrix :math:`A` in the objective function above.
+
+    B : 2d-array, square
+        The square matrix :math:`B` in the objective function above.
+
+    method :  str in {'faq', '2opt'} (default: 'faq')
+        The algorithm used to solve the problem. This is the method-specific
+        documentation for 'faq'.
+        :ref:`'2opt' <optimize.qap-2opt>` is also available.
+
+    Options
+    -------
+
+    maximize : bool (default = False)
+        Setting `maximize` to ``True`` solves the Graph Matching Problem (GMP)
+        rather than the Quadratic Assingnment Problem (QAP). This is
+        accomplished through trivial negation of the objective function.
+
+    rng : {None, int, `~np.random.RandomState`, `~np.random.Generator`}
+        This parameter defines the object to use for drawing random
+        variates.
+        If `rng` is ``None`` the `~np.random.RandomState` singleton is
+        used.
+        If `rng` is an int, a new ``RandomState`` instance is used,
+        seeded with `rng`.
+        If `rng` is already a ``RandomState`` or ``Generator``
+        instance, then that object is used.
+        Default is None.
+
+    partial_match : 2d-array of integers, optional, (default = None)
+        Allows the user to fix part of the matching between the two
+        matrices. In the literature, a partial match is also known as a
+        "seed".
+
+        Each row of `partial_match` specifies the indices of a pair of
+        corresponding nodes, that is, node ``partial_match[i, 0]`` of `A` is
+        matched to node ``partial_match[i, 1]`` of `B`. Accordingly,
+        ``partial_match`` is an array of size ``(m , 2)``, where ``m`` is
+        not greater than the number of nodes.
+
+    P0 : 2d-array, "barycenter", or "randomized" (default = "barycenter")
```

Comment:
`scipy.optimize.differential_evolution` uses `"random"` rather than `"randomized"`.
Also `scipy.cluster.vq.kmeans2`.

Rename it for consistency?

Reply:
either works for me. @mdhaber thoughts?

Reply:
@pmla `differential_evolution` calls the parameter `init`, and similarly, `scipy.cluster.vq.kmeans2` calls the parameter `minit`. Since you asked for the argument to be named `P0` instead of `init`, shouldn't we at least be consistent in breaking consistency with those functions?

Actually, @asaadeldin11 had `random` in the initial commit, and I suggested `randomized` to emphasize that it was not entirely random, but a randomized variant of the barycenter. Ultimately he made the decision between the two. I can see it going either way, but as with so many of these things, I think a more objective reason than partial consistency would be needed to ask him to change it back again. Do you agree?

Reply:
fwiw, whatever @pmla and @mdhaber think is best for the scientific community is what i would request @asaadeldin11 do.  he will be fine either way 👍

Reply:
A tangent: I have made a blunder throughout this PR. I have not previously worked with PRs which themselves have PRs, as opposed to direct commits to the PR fork with discussion on the main thread. I did not realize that there was a lot of communication in the now-closed PRs on your scipy fork. I apologize for the many annoying suggestions on issues which you had already agreed upon.

Reply:
I have no issue making the change again, whatever you all think is best for the function is fine with me.

Reply:
After reading your discussion I think it is fine as-is.